### PR TITLE
[ROCm]Enable AITER MoE backend for MiniMax-M3-MXFP4

### PR DESCRIPTION
**Pull request by qli88**
[ROCm][feature] Enable MiniMax-M3-MXFP4 with AITER MoE

This feature requires AITER version bump (latest version). 

**Accuracy test:**
1. vLLM server start: VLLM_ROCM_USE_AITER=1 VLLM_ROCM_USE_AITER_MOE=1 VLLM_USE_BREAKABLE_CUDAGRAPH=0 vllm serve  /data/amd-MiniMax-M3-MXFP4/   --block-size 128   -tp 8   --attention-backend TRITON_ATTN   --tool-call-parser minimax_m3   --enable-auto-tool-choice   --reasoning-parser minimax_m3 --moe-backend aiter
2. gsm8k: 
  lm_eval \
  --model local-chat-completions \
  --model_args "model=/data/amd-MiniMax-M3-MXFP4/,base_url=http://127.0.0.1:8000/v1/chat/completions,num_concurrent=32,max_gen_toks=16384" \
  --tasks gsm8k \
  --num_fewshot 5 \
  --batch_size 1 \
  --apply_chat_template \
  --fewshot_as_multiturn

|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9401|±  |0.0065|
|     |       |strict-match    |     5|exact_match|↑  |0.9409|±  |0.0065|

### CODE REVIEW

File: vllm/models/minimax_m3/amd/model.py [hunk 1]
```diff
@@ -305,6 +305,7 @@ def __init__(
             top_k=config.num_experts_per_tok,
             hidden_size=config.hidden_size,
             intermediate_size=config.intermediate_size,
+            intermediate_pad=0,
```
Comment (BowenBao) [inline vllm/models/minimax_m3/amd/model.py:312]: why is this special case necessary for minimax?

**Comment by Fangzhou-Ai** [comment]
Do we have any perf gain regarding this PR?

File: vllm/models/minimax_m3/amd/model.py (hunk 1 above)
Comment (qli88) [reply vllm/models/minimax_m3/amd/model.py:312]: This is to avoid 'pad-skipping' accuracy issue in this version of AITER. I will submit an issue on AITER side and after it is resolved, I will remove this.

**Comment by Fangzhou-Ai** [comment]
Update my perf result on 8 MI355X
ISL/OSL 8k1k conc=4
```
============ Serving Benchmark Result ============
Successful requests:                     40        
Benchmark duration (s):                  86.71     
Total input tokens:                      294213    
Total generated tokens:                  36805     
Request throughput (req/s):              0.46      
Output token throughput (tok/s):         424.44    
Total Token throughput (tok/s):          3817.31   
---------------Time to First Token----------------
Mean TTFT (ms):                          274.54    
Median TTFT (ms):                        256.45    
P90 TTFT (ms):                           272.89    
P99 TTFT (ms):                           603.19    
P99.9 TTFT (ms):                         677.08    
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          8.92      
Median TPOT (ms):                        8.97      
P90 TPOT (ms):                           9.11      
P99 TPOT (ms):                           9.15      
P99.9 TPOT (ms):                         9.16      
---------------Inter-token Latency----------------
Mean ITL (ms):                           8.92      
Median ITL (ms):                         8.34      
P90 ITL (ms):                            8.39      
P99 ITL (ms):                            8.51      
P99.9 ITL (ms):                          226.08    
----------------End-to-end Latency----------------
Mean E2EL (ms):                          8468.78   
Median E2EL (ms):                        8354.85   
P90 E2EL (ms):                           9269.13   
P99 E2EL (ms):                           9645.25   
P99.9 E2EL (ms):                         9683.83   
==================================================
```

**Comment by Fangzhou-Ai** [comment]
This PR is NOT ready to merge @tjtanaa  until aiter version is bumped to v0.1.15.post3

File: vllm/model_executor/layers/fused_moe/config.py [hunk 2]
```diff
@@ -1277,6 +1277,9 @@ class FusedMoEConfig:
     hidden_dim_unpadded: int | None = None
     # Defaults to intermediate_size_per_partition if not specified.
     intermediate_size_per_partition_unpadded: int | None = None
+    # Model specific override
```
Comment (tjtanaa) [inline vllm/model_executor/layers/fused_moe/config.py:1282]: We already have this padding and unpadding logic. How different is this padding frome the existing field?
@qli88 

@BowenBao I remember AMD added this arguments for GPTOSS and is not used by other platform. So will need an explanation about the need to add more arguments. 

CC @dllehr-amd

File: vllm/model_executor/layers/fused_moe/experts/rocm_aiter_moe.py [hunk 3]
```diff
@@ -4,6 +4,7 @@
 from functools import lru_cache
 
 import torch
+from aiter.ops.flydsl.moe_common import GateMode
```
Comment (tjtanaa) [inline vllm/model_executor/layers/fused_moe/experts/rocm_aiter_moe.py]: Another thing, any aiter import should be inside a function scope to avoid import error on other platforms.

**Comment by fxmarty-amd** [comment]
Along with `EMULATION` backend, the AITER MXFP4 MOE backend is the only correct one to run https://huggingface.co/amd/MiniMax-M3-MXFP4, so this PR is quite important.

The current backend `TRITON_UNFUSED` does not apply activation QDQ, so it does not really apply `config.json` that requests activation quantization. It could be supported in the future (maybe in https://github.com/vllm-project/vllm/pull/46491)

File: vllm/model_executor/layers/fused_moe/config.py (hunk 2 above)
Comment (BowenBao) [reply vllm/model_executor/layers/fused_moe/config.py:1282]: afaik the "xxx_unpadded" args was genuine and not GPTOSS only. Padding is essential to cover any model with shape not divisible by group size / tp size. It wasn't AMD specific as well.

That said this is good point, the pad value can be calculated from `hidden_dim` - `hidden_dim_unpadded`, so seems redundant.

Comment (qli88) [reply vllm/model_executor/layers/fused_moe/config.py:1282]: This is to provide a model specific padding size to override "hidden_dim - hidden_dim_unpadded". This very case is to circumvent an AITER issue which will cause accuracy issue when performing pad-skipping calculations. I will create an issue on AITER side and will revisit this after that issue gets resolved.

File: vllm/model_executor/layers/fused_moe/layer.py [hunk 4]
```diff
@@ -105,6 +105,8 @@ def FusedMoE(
     top_k: int,
     hidden_size: int,
     intermediate_size: int,
+    hidden_pad: int | None = None,
```
Comment (tjtanaa) [inline vllm/model_executor/layers/fused_moe/layer.py]: Adding the argument at this place might break at places that does not uses keyword argument. I have triggered the CI to validate this and I will check the source code as well.

File: vllm/models/minimax_m3/amd/model.py (hunk 1 above)
Comment (tjtanaa) [reply vllm/models/minimax_m3/amd/model.py:312]: @qli88 `hidden_pad` is not used at all. Let's not include it. Just introduce `intermediate_pad` will do.

File: vllm/model_executor/layers/fused_moe/layer.py (hunk 4 above)
Comment (tjtanaa) [reply vllm/model_executor/layers/fused_moe/layer.py]: From the source code, it seems every call side of `FusedMoE` are passing keyword arguments. Most likely it is safe. Let's check CI

File: vllm/model_executor/layers/fused_moe/experts/rocm_aiter_moe.py [hunk 5]
```diff
@@ -357,14 +370,17 @@ def rocm_aiter_fused_experts(
         # `rocm_aiter_ops.shuffle_weight_a16w4` in `oracle/mxfp4.py`,
         # which always sets `is_guinterleave=True`.
         # Hence, we pass in GateMode.INTERLEAVE to match the weight shuffling.
+        from aiter.ops.flydsl.moe_common import GateMode
```
Comment (tjtanaa) [inline vllm/model_executor/layers/fused_moe/experts/rocm_aiter_moe.py:370]: This PR can only be merged after we upgrade AITER. current AITER v0.1.13.post1 does not have this import path. It will fail.

File: vllm/model_executor/layers/fused_moe/config.py [hunk 6]
```diff
@@ -1277,6 +1277,9 @@ class FusedMoEConfig:
...
+    hidden_pad: int | None = None
```
Comment (tjtanaa) [inline vllm/model_executor/layers/fused_moe/config.py]: hidden_pad is not used at all. Let's not include it. Just introduce intermediate_pad will do.

File: vllm/model_executor/layers/fused_moe/layer.py (hunk 4 above)
Comment (tjtanaa) [reply vllm/model_executor/layers/fused_moe/layer.py]: hidden_pad is not used at all. Let's not include it. Just introduce intermediate_pad will do.

**Review by tjtanaa — APPROVED** [review]
LGTM. Only merged once aiter version is upgraded

File: vllm/model_executor/layers/fused_moe/config.py (hunk 2 above)
Comment (Rohan138) [reply vllm/model_executor/layers/fused_moe/config.py:1282]: FYI for NV the padding is handled by the FI kernels, see: https://github.com/vllm-project/vllm/pull/30647; AITER handles padding for some backends (FlyDSL and Triton) but not CK/CKTile, would be nice to have all of it handled by AITER and remove the AMD-specific padding bits from vLLM
